### ISSUE TITLE
How to compute the value 'test_iter'

### ISSUE THREAD

**Issue by Xiao-G1023** [yjxiong/temporal-segment-networks#55]
Hi Mr. Xiong and Wang! @yjxiong @wanglimin 
In **models/hmdb51/tsn_bn_inception_rgb_solver.prototxt**, how do you get the value of `test_iter: 383 ` ? 
In **models/hmdb51/tsn_bn_inception_rgb_train_val.prototxt**, we can find the Test parameter:
 layer {

  name: "data"
  type: "VideoData"
  top: "data"
  top: "label"
  video_data_param {
    **source: "data/hmdb51_rgb_val_split_1.txt"**
    **batch_size: 1**
    new_length: 1
    num_segments: 3
    modality: RGB
    name_pattern: "img_%05d.jpg"
  }
  transform_param{
    crop_size: 224
    mirror: false
    mean_value: [104, 117, 123, 104, 117, 123, 104, 117, 123]
  }
  include: { phase: TEST }
}

In the obtained file **data/hmdb51_rgb_val_split_1.txt**, the total number of test-videos is 1530.
**It really confused me about the value of test_iter.**
When I fine-tune TSN, it does not work and tells the following errors:
F0224 14:23:00.519636 15726 reshape_layer.cpp:79] Check failed: 0 == bottom[0]->count() % explicit_count (0 vs. 192) bottom count (192) must be divisible by the product of the specified dimensions (303)

*** Check failure stack trace: ***
    @     0x7f256daeddaa  (unknown)
    @     0x7f256daedce4  (unknown)
    @     0x7f256daed6e6  (unknown)
    @     0x7f256daf0687  (unknown)
    @     0x7f256dedea0f  caffe::ReshapeLayer<>::Reshape()
    @     0x7f256df6b360  caffe::Net<>::Init()
    @     0x7f256df6e0b3  caffe::Net<>::Net()
    @     0x7f256df80a45  caffe::Solver<>::InitTrainNet()
    @     0x7f256df80fa4  caffe::Solver<>::Init()
    @     0x7f256df813d5  caffe::Solver<>::Solver()
    @           0x41cbe6  caffe::SGDSolver<>::SGDSolver()
    @           0x41cd88  caffe::GetSolver<>()
    @           0x413128  train()
    @           0x410fdd  main
    @     0x7f256c5a7f45  (unknown)
    @           0x4113de  (unknown)
    @              (nil)  (unknown)
Setting device 0

### Is that because of a wrong number of `test_iter`?
Thanks for your attention!!!

**Comment by yjxiong** [comment]
`test_iter` is computed by `total_sample_num` / `batch_size` / `iter_size` / `num_gpu`.

However, the error you list is not related to `test_iter`. If you look at your log, you will find that it is related to `reshape` layer. So you may have some shape mismatch instead of anything to do with `test_iter`.

**Comment by yjxiong** [comment]
One thing to add. This on-the-fly validation is just for monitoring the training. All experimental results should be reported with the provide testing scripts.

**Comment by Xiao-G1023** [comment]
Thanks for you reply!! @yjxiong 
Now I know how to set the parameter `test_iter`! Really appraciate you!
**About the above error in` reshape layer`, I still don't konw which point I miss.** 
I extract frames and optical flow images using the `dense_flow` libraries. As the data is 2 classes, I make the following changes:
(1) change the "InnerProduct" layer of the **train_val_rgb.protetxt**:   `name:fc-action-new`;  ` num_output:2`;  `param{}`
(2) change the **solver_rgb.prototxt**: `test_iter:  155`; `base_lr: 0.001`

Where should I focus on?? Could you give me some advice? 
Thanks you!

**Comment by Xiao-G1023** [comment]
HA! I have find the missing point. In the layer
` layer { name: "reshape_fc" type: "Reshape" bottom: "fc" top: "reshape_fc" reshape_param { shape { dim: [-1, 1, 3, 101] } } }`, 
the last parameter is 101, which I didn't change the corresponding classes 2.
Now the fine-tuned TSN is working well.
Thank you all the same!!

**Comment by yjxiong** [comment]
Generally, you should use the provided init models for initialization instead of the models trained on UCF101/HMDB51. The high gradient norm is usually due to increase `iter_size`, which is fine. You can increase the value of `clip_gradient` in your solver to fit it.
